include/constants/battle_config.h

```diff
@@ -123,6 +123,8 @@
 // Item settings
 #define B_HP_BERRIES                GEN_6 // In Gen4+, berries which restore hp activate immediately after hp drops to half. In gen3, the effect occurs at the end of the turn.
 #define B_BERRIES_INSTANT           GEN_6 // In Gen4+, most berries activate on battle start/switch-in if applicable. In gen3, they only activate either at the move end or turn end.
+#define X_ITEMS_REWORK              FALSE // This flag changes the way in which X Items' effects are implemented in the code and relies on the item_expansion to work.
```

Comment:
is it possible to use `#ifdef ITEM_EXPANSION` rather than needing a new `X_ITEMS_REWORK`?

Reply:
I'm not sure that's a good idea since this constant label is what allows the rework to work, and the changes or at least some of the changes that are built into a ROM if it's set to `TRUE` shouldn't really depend on the item_expansion :eyes: ...

Reply:
I tested using the ifdef method and it worked wonderfully where you replace `#if X_ITEMS_REWORK == TRUE` with `#ifdef ITEM_EXPANSION` and vise versa. This also removes the need to have the `#if` check for `X_ITEMS_REWORK` in #1277.

The reason that this is a much cleaner solution is because it removes this awkward toggle option where if it's set to `TRUE` with only BE it prevents the X Items from working completely and if it's set to `FALSE` with the current state of the PR and #1277 + IE merged, X Sp. Def will be usable, but not work correctly. These are very awkward situations of functionality for a user to potentially to run into if they have control over this setting.

The Rework always being used when IE is merged and not being used when not merged allows X Items to always be usable as expected.